### PR TITLE
recheck current route guards on permission change

### LINKED ISSUE #1439: Permission revocation of navigation menu items requires forced reload or opening of another menu point to be active for users on their device
If a user's permission to see a menu item is revoked, but the user has opened the menu, the content of the menu will still be displayed in the user's browser. Only the menu item disappears from the navigation bar. Only when the user changes the menu or reloads the page he has no further access.

Reproduction: 
1. Two Accounts are needed; Superadmin und normal user account. Normal should be group staff. 
1. open meeting with both accounts in different browsers
2. navigate with the normal user account to the participant menu
3. open as superadmin the participant menu. open the three-dot-menu and choose groups. revoke permission for group staff to see other participants
4. check the normal user view. menu is gone, but the user still sees the content.

What should happen: When the permission to see a menu item is revoked, there should be an automatic redirection to the first available menu item. Example: If available, then redirect to home page, otherwise redirect to next available menu item.

### PR DESCRIPTION
**Pull request by bastianjoel**
resolves #1439

### CODE REVIEW

**Review by luisa-beerboom — APPROVED** [review]

**Review by luisa-beerboom — APPROVED** [review]

File: client/src/app/site/guards/permission.guard.ts [hunk 1]
```diff
@@ -13,26 +14,25 @@ export class PermissionGuard implements CanLoad {
...
     ) {}
 
-    public async canLoad(route: Route, segments: UrlSegment[]): Promise<boolean> {
+    public async canLoad(route: Route, segments: UrlSegment[]): Promise<boolean | UrlTree> {
         const url = this.getCurrentNavigationUrl();
         if (this.osRouter.isOrganizationUrl(url)) {
             if (!(await this.authCheck.isAuthorizedToSeeOrganization())) {
-                this.reroute.forwardToOnlyMeeting(url === `/info` ? [`info`] : []);
-                return false;
+                const meetingId = this.operator.onlyMeeting.toString();
```
Comment (luisa-beerboom) [inline client/src/app/site/guards/permission.guard.ts:25]: Why not continue using forwardToOnlyMeeting here?
Haven't you changed it to do exactly what you are also doing here?

File: client/src/app/site/services/reroute.service.ts [hunk 2]
```diff
@@ -63,16 +65,16 @@ export class RerouteService {
     /**
      * If the user requested a route without direct meaning, forward to their meaningful meeting
      */
-    public forwardToOnlyMeeting(segments: string[] = []): void {
+    public forwardToOnlyMeeting(segments: string[] = []): UrlTree | void {
```
Comment (luisa-beerboom) [inline client/src/app/site/services/reroute.service.ts:68]: With your changes to take out the navigation most of the `RerouteService`'s method names don't fit anymore, please change names like `forwardToOnlyMeeting` that insinuate that they also handle the navigation to something more fitting like maybe `getOnlyMeetingUrlTree` (please don't forget to change the documentation either)

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]

**Review by luisa-beerboom — APPROVED** [review]

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
Go as normal user to e.g. agenda list.
Now remove agenda.can_see permission.
-> User gets Authorization Error for agenda_item.can_see (OK)
Now click on "Back to OpenSlides" button which shows again a 2nd Authorization Error (for meeting.can_see_frontpage).
User has to click 2-3 times on "Back to OpenSlides" to go back to frontpage.

**Comment by bastianjoel** [comment]
I can't replicate this. Anyway I've changed the way the error page redirects and hope this fixes the issue.

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
same problem, reproduceable for me

**Comment by bastianjoel** [comment]
I can reproduce this now. Happens only in chromium.

**Review by luisa-beerboom — APPROVED** [review]

**Review by luisa-beerboom — APPROVED** [review]

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
Retested: Same error.
Get "Error: Uncaught (in promise): [object Undefined]" after clicking 2-3 times on "Back to OpenSlides"
Tested with Chromium and Firefox.

**Review by emanuelschuetze — APPROVED** [review]
Looks good now!

**Pull request merged**